client/network/src/protocol.rs

```diff
@@ -808,6 +820,19 @@ impl<B: BlockT> Protocol<B> {
 			}
 		}
 
+		if status.roles.is_full() && self.sync.num_peers() >= self.default_peers_set_num_full {
+			debug!(target: "sync", "Too many full nodes, rejecting {}", who);
+			self.behaviour.disconnect_peer(&who, HARDCODED_PEERSETS_SYNC);
+			return Err(())
+		} else if status.roles.is_light() &&
```

Comment:
This else is redundant, please get rid of it. I love that you used `roles.is_light()` here, although it is actually always `!roles.is_full()`, but having it made it easier to read the code. Why are we not worried here about overflow?

Reply:
`self.peers` contains both full and light peers, but `self.sync` only contains full peers, so the subtraction should always succeed